### ISSUE TITLE
Method call getPasswordCredentials of Application entity throws Null pointer exception

### ISSUE THREAD

**Issue by sushanthapoojary** [davidmoten/odata-client#28]
Hi David,

When I upgraded library to 0.1.20 version, I observed method call getPasswordCredentials of Application entity is failing. It is throwing null pointer exception.
In earlier version(0.1.19) it was working as expected by returning CollectionPage<PasswordCredential>. 

Sample code:
List<PasswordCredential> passwordCredentialList = new ArrayList<>();
passwordCredentialList.add(PasswordCredential.builder().secretText("Application secret Text").build());

Application application = Application.builderApplication().passwordCredentials(passwordCredentialList).build();

//Throws Null Pointer Exception in 0.1.20 version of library
application.getPasswordCredentials();

Is it failing because of recent changes on handling null List by CollectionPage?

**Comment by davidmoten** [comment]
Thanks for the report. Yep I added a Precondition in 0.1.20 which was not necessary when nextLink is null so I'll fix that now and release.

**Comment by davidmoten** [comment]
By the way, in release about to be built I've added varargs overloads to the builders so you don't need to build Lists like that. You can just go:

```java
PasswordCredential pc1= PasswordCredential.builder().secretText("Application secret Text").build();
PasswordCredential pc2= PasswordCredential.builder().secretText("Another secret Text").build();
Application application = Application.builderApplication().passwordCredentials(pc1, pc2).build();
```
or in your example just:
```java
PasswordCredential pc= PasswordCredential.builder().secretText("Application secret Text").build();
Application application = Application.builderApplication().passwordCredentials(pc).build();

```

**Comment by davidmoten** [comment]
fixed in release 0.1.22 on Maven Central now

**Comment by sushanthapoojary** [comment]
Thanks a lot for quick turnaround. 
It's nice to have PasswordCredential builder. Thanks again for your extra effort to add this.
